smarts/env/gymnasium/hiway_env_v1.py

```diff
@@ -0,0 +1,390 @@
+
+        # TODO: set observation space
+        self.observations_formatter = ObservationsSpaceFormat(agent_interfaces)
+        self.observation_space = self.observations_formatter.space
+
+        from smarts.core.smarts import SMARTS
+
+        self._smarts = SMARTS(
+            agent_interfaces=agent_interfaces,
+            traffic_sims=traffic_sims,
+            envision=visualization_client,
+            visdom=visdom_client,
+            fixed_timestep_sec=fixed_timestep_sec,
+            zoo_addrs=zoo_addrs,
+        )
+
+    def step(
+        self, action: ActType
+    ) -> Tuple[ObsType, SupportsFloat, bool, bool, Dict[str, Any]]:
+        """Run one timestep of the environment's dynamics using the agent actions.
+        When the end of an episode is reached (``terminated or truncated``), it is necessary to call :meth:`reset` to
+        reset this environment's state for the next episode.
+        .. versionchanged:: 0.26
+            The Step API was changed removing ``done`` in favor of ``terminated`` and ``truncated`` to make it clearer
+            to users when the environment had terminated or truncated which is critical for reinforcement learning
+            bootstrapping algorithms.
+        Args:
+            action (ActType): an action provided by the agent to update the environment state.
+        Returns:
+            observation (ObsType): An element of the environment's :attr:`observation_space` as the next observation due to the agent actions.
+                An example is a numpy array containing the positions and velocities of the pole in CartPole.
+            reward (SupportsFloat): The reward as a result of taking the action.
+            terminated (bool): Whether the agent reaches the terminal state (as defined under the MDP of the task)
+                which can be positive or negative. An example is reaching the goal state or moving into the lava from
+                the Sutton and Barton, Gridworld. If true, the user needs to call :meth:`reset`.
+            truncated (bool): Whether the truncation condition outside the scope of the MDP is satisfied.
+                Typically, this is a timelimit, but could also be used to indicate an agent physically going out of bounds.
+                Can be used to end the episode prematurely before a terminal state is reached.
+                If true, the user needs to call :meth:`reset`.
+            info (dict): Contains auxiliary diagnostic information (helpful for debugging, learning, and logging).
+                This might, for instance, contain: metrics that describe the agent's performance state, variables that are
+                hidden from observations, or individual reward terms that are combined to produce the total reward.
+                In OpenAI Gym <v26, it contains "TimeLimit.truncated" to distinguish truncation and termination,
+                however this is deprecated in favour of returning terminated and truncated variables.
```

Comment:
```suggestion
            info (dict): Contains auxiliary diagnostic information (helpful for debugging, learning, and logging).
                This might, for instance, contain: metrics that describe the agent's performance state, variables that are
                hidden from observations, or individual reward terms that are combined to produce the total reward.
```